tests/test_bundle.py

```diff
@@ -112,37 +96,70 @@ async def test_build_deploy_knative_charms(ops_test: OpsTest):
     )
 
 
-async def test_cloud_events_player_example(ops_test: OpsTest):
-    await ops_test.run(
-        "kubectl",
-        "apply",
-        "-f",
-        "./examples/cloudevents-player.yaml",
-        check=True,
-    )
-    await ops_test.run(
-        "kubectl",
-        "wait",
-        "--for=condition=ready",
-        "ksvc",
-        "cloudevents-player",
-        "--timeout=5m",
-        check=True,
+RETRY_FOR_MINUTE = Retrying(
+    stop=stop_after_delay(60),
+    wait=wait_fixed(5),
+    reraise=True,
+)
+
+
+@pytest.fixture()
+def wait_for_ksvc():
+    """Waits until KNative Serving Service is available, to a maximum 1 minute"""
+    lightkube_client = Client()
+    for attempt in RETRY_FOR_MINUTE:
+        with attempt:
+            ksvcs = list(lightkube_client.list(KSVC))
+
+
+@pytest.fixture()
+def remove_cloudevents_player_example():
+    """Fixture that attempts to remove the cloudevents-player example after a test has run"""
+    yield
+    lightkube_client = Client()
+    lightkube_client.delete(KSVC, "cloudevents-player")
+
+
+def wait_for_ready(resource, name, namespace, timeout=60, wait=5):
```

Comment:
I'm wondering why you are using this instead of throwing the wait to `tenacity`

Reply:
Yeah I debated that too.  Tenacity is centered around things that either succeed or throw an exception - since this wasn't throwing, I went a manual route.  But it does feel odd since tenacity is used elsewhere.

Would you prefer refactoring to use tenacity and then to throw an error if the ksvc is not ready (that way tenacity is happy and retries)?

Reply:
Yes, you can use an `assert` if you don't want to throw an error manually. [We did that for training-operator](https://github.com/canonical/training-operator/blob/main/tests/integration/test_charm.py#L115).